### PR TITLE
Filter out invalid characters at user creation

### LINKED ISSUE #990: user name validation too strict
rule for valid characters in `userFirstName`, `userLastName`:

~~`allowed c = ...`~~ everything goes!

login nicks will be composed of small letters only; special characters are ignored.

~~(not sure: what's the current behavior?)~~ current behavior: a-okay except the login construction does not filter invalid characters.  see https://github.com/liqd/aula/issues/990#issuecomment-254502893 below.

### PR DESCRIPTION
**Pull request by andorp**
Fixes #990

### CODE REVIEW

**Comment by andorp** [comment]
@fisx: Could you take a look at the failing test case.

File: tests/Persistent/PureSpec.hs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+sampleUsernameSize :: Int
+sampleUsernameSize = 5
+
+spec :: Spec
+spec = do
+    describe "genUserLoginFromRealname" $ do
+
+        let checkValidUsernames  = all (ST.all usernameAllowedChar)
+            checkValidUsernames' = all (isRight . testValidator (usernameV :: FieldValidator ST String))
+            validInvalidChars   = oneof
```
Comment (fisx) [inline tests/Persistent/PureSpec.hs]: which one?  valid or invalid?

Comment (andorp) [inline tests/Persistent/PureSpec.hs]: and

**Comment by fisx** [comment]
@andorp please review and merge if you and travis are green.

thanks!

**Pull request merged**